### PR TITLE
Allow to enable feature gates

### LINKED ISSUE #1068: Allow to enable feature gates
**What would you like to be added**:

Possibility to enable feature gates via `cluster.yml`.

**Example cluster.yml**:

```yaml
api:
  feature_gates: {}
kubelet:
  feature_gates: {}
```

### PR DESCRIPTION
**Pull request by nevalla**
This PR allows to configure feature gates in `cluster.yml`
```
...
control_plane:
  feature_gates:
    featureA: true
kubelet:
  feature_gates:
    featureA: true
...
```
feature gates for **control_plane** are passed as extraArgs for `apiServer`, `scheduler` and `controllerManager`  kubeadm components.

`kubeadm-reconfigure.sh` uploads now kubeadm config to cluster and new phase `ReconfigureKubelet` that will read kubelet config from cluster and reloads kubelet if needed.

fixes #1068

### CODE REVIEW

File: lib/pharos/config_schema.rb [hunk 1]
```diff
@@ -94,6 +94,7 @@ def self.messages
         optional(:name).filled(:str?)
         optional(:api).schema do
           optional(:endpoint).filled(:str?)
+          optional(:feature_gates).filled
```
Comment (jnummelin) [inline lib/pharos/config_schema.rb]: IMO this should be moved under `control_plane` key as these feature gates will now be set on all control plane components.

Comment (nevalla) [reply lib/pharos/config_schema.rb]: Maybe the best solution is to define config options for all components.

Comment (jnummelin) [reply lib/pharos/config_schema.rb]: If we can safely propagate all the control plane feature gates to all control plane components, assuming that a component will just ignore a feature it does not understand, I think having them defined on `control_plane` level makes users lifes so much easier.

Comment (nevalla) [reply lib/pharos/config_schema.rb]: oh, we already have `control_plane` option

Comment (nevalla) [reply lib/pharos/config_schema.rb]: moved api feature gates under `control_plane`

**Comment by jnummelin** [comment]
Overall looks pretty good. I didn't do any real testing though yet.

File: lib/pharos/kubeadm/kubelet_config.rb [hunk 2]
```diff
@@ -30,6 +30,9 @@ def generate
         if @config.kubelet&.read_only_port
           config['readOnlyPort'] = 10_255
         end
+        if @config.kubelet&.feature_gates
```
Comment (jakolehm) [inline lib/pharos/kubeadm/kubelet_config.rb]:
```suggestion
        if feature_gates = @config.kubelet&.feature_gates
```

File: lib/pharos/kubeadm/kubelet_config.rb [hunk 3]
```diff
@@ -30,6 +30,9 @@ def generate
...
+          config['featureGates'] = @config.kubelet.feature_gates
```
Comment (jakolehm) [inline lib/pharos/kubeadm/kubelet_config.rb]:
```suggestion
          config['featureGates'] = feature_gates
```

**Pull request merged**